plugins/MultiSites/DataTable/Filter/NestedSitesLimiter.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\MultiSites\DataTable\Filter;
+
+use Piwik\DataTable\BaseFilter;
+use Piwik\DataTable\Row;
+use Piwik\DataTable;
+
+/**
+ * Makes sure to not have any subtables anymore and applies the limit to the flattened table.
+ *
+ * So if $table is
+ * array(
+ *    site1
+ *    group2
+ *        subtable => site3
+ *                    site4
+ *                    site5
+ *    site6
+ *    site7
+ * )
+ *
+ * it will format this to
+ *
+ * array(
+ *    site1
+ *    group2
+ *    site3
+ *    site4
+ *    site5
+ *    site6
+ *    site7
+ * )
+ *
+ * and then apply the limit filter.
+ *
+ * Each group will not count into the limit/offset. This way, if one requests a limit of 50 sites,
+ * we make sure to return 50 sites.
+ *
+ * @param $sites
+ * @return array
+ */
+class NestedSitesLimiter extends BaseFilter
+{
+    private $offset = 0;
+    private $limit  = 0;
+    /**
+     * @var Row[]
+     */
+    private $rows   = array();
+
+    /**
+     * Constructor.
+     *
+     * @param DataTable $table The table to eventually filter.
+     */
+    public function __construct($table, $offset, $limit)
+    {
+        parent::__construct($table);
+        $this->offset = $offset;
+        $this->limit  = $limit;
+    }
+
+    /**
+     * @param DataTable $table
+     */
+    public function filter($table)
+    {
+        $numRows = 0;
+        $lastGroupFromPreviousPage = null;
+
+        foreach ($table->getRows() as $row) {
+
+            $this->addRowIfNeeded($row, $numRows);
+            $numRows++;
+
+            $subtable = $row->getSubtable();
+            if ($subtable) {
+                if (!$this->hasRows()) {
+                    $lastGroupFromPreviousPage = $row;
+                }
+                foreach ($subtable->getRows() as $subRow) {
+                    $this->addRowIfNeeded($subRow, $numRows);
+                    $numRows++;
+                }
+                $row->removeSubtable();
+            }
+
+            if ($this->hasNumberOfRequestedRowsFound()) {
+                break;
+            }
+        }
+
+        $this->prependGroupIfFirstSiteBelongsToAGroupButGroupIsMissingInRows($lastGroupFromPreviousPage);
```

Comment:
This is another bugfix. In previous version if we had this result:

* Site 5 (belongs to "group 2")
* Site 6
* Site 7

we did not prepend "Group2". Now we return 

* Group 2
* Site 5 
* Site 6
* Site 7

Even if only 3 sites were requested. This is expected and the dashboard worked like this before in 2.12.